lib/mayaUsd/render/MaterialXGenOgsXml/GlslFragmentGenerator.cpp

```diff
@@ -25,6 +27,74 @@ const string LIGHT_LOOP_RESULT = "lightLoopResult";
 const string MAYA_ENV_IRRADIANCE_SAMPLE = "diffuseI";
 const string MAYA_ENV_RADIANCE_SAMPLE = "specularI";
 const string MAYA_ENV_ROUGHNESS = "roughness";
+
+// The Apple shader compiler found on M1/M2 machines does not allow using a global variable as a
+// temporary buffer. We will need to handle these inputs differently.
+void fixupVertexDataInstance(ShaderStage& stage)
+{
+    // All the strings start with $. This is an anchor character in regex, so skip it.
+    auto d = [](auto const& s) { return s.substr(1); };
+
+    // Find keywords:                   (as text)
+    //
+    //  vec3 HW::T_POSITION_WORLD       vec3 $normalWorld
+    //
+    // And replace with:
+    //
+    // vec3 unused_T_POSITION_WORLD     vec3 unused_normalWorld
+    //
+
+    static const std::string paramSource = "vec3 [$](" + d(HW::T_POSITION_WORLD) + "|"
+        + d(HW::T_POSITION_OBJECT) + "|" + d(HW::T_NORMAL_WORLD) + "|" + d(HW::T_NORMAL_OBJECT)
+        + "|" + d(HW::T_TANGENT_WORLD) + "|" + d(HW::T_TANGENT_OBJECT) + "|"
+        + d(HW::T_BITANGENT_WORLD) + "|" + d(HW::T_BITANGENT_OBJECT) + ")";
+
+    static const std::regex paramRegex(paramSource.c_str());
+
+    // Find keywords:                                       (as text)
+    //
+    //  HW::T_VERTEX_DATA_INSTANCE.HW::T_POSITION_WORLD     $vd.$normalWorld
+    //
+    // And replace with:
+    //
+    // HW::T_POSITION_WORLD(PIX_IN.HW::T_POSITION_WORLD)    $normalWorld( PIX_IN.$normalWorld )
+    //
+
+    static const std::string vtxSource = "[$]" + d(HW::T_VERTEX_DATA_INSTANCE) + "[.][$]("
+        + d(HW::T_POSITION_WORLD) + "|" + d(HW::T_POSITION_OBJECT) + "|" + d(HW::T_NORMAL_WORLD)
+        + "|" + d(HW::T_NORMAL_OBJECT) + "|" + d(HW::T_TANGENT_WORLD) + "|"
+        + d(HW::T_TANGENT_OBJECT) + "|" + d(HW::T_BITANGENT_WORLD) + "|" + d(HW::T_BITANGENT_OBJECT)
+        + ")";
+
+    static const std::regex vtxRegex(vtxSource.c_str());
+
+    // Find keywords:                   (as text)
+    //
+    //  HW::T_VERTEX_DATA_INSTANCE.     $vd.$inGeomprop_st
+    //
+    // And replace with:
+    //
+    //  (nothing)                       $inGeomprop_st
```

Comment:
These are captured in the shader main and passed as parameters, so they do not require the `PIX_IN` treatment.